nmdc_runtime/api/endpoints/wf_file_staging.py

```diff
@@ -114,3 +118,97 @@ def update_globus_tasks(
         {"task_id": task_id}, doc_globus_patched
     )
     return doc_globus_patched
+
+
+@router.get(
+    "/wf_file_staging/globus_tasks",
+    response_model=ListResponse[GlobusTask],
+    response_model_exclude_unset=True,
+)
+def list_globus_tasks(
+    req: Annotated[ListRequest, Query()],
+    mdb: Database = Depends(get_mongo_db),
+    user: User = Depends(get_current_active_user),
+):
+    # check for permissions first
+    check_can_run_wf_file_staging_endpoints(user)
+
+    return list_resources(req, mdb, "wf_file_staging.globus_tasks")
+
+
+@router.post(
+    "/wf_file_staging/jgi_samples",
+    status_code=status.HTTP_201_CREATED,
+    response_model=JGISample,
+)
+def create_jgi_samples(
+    jgi_in: JGISample,
+    mdb: Database = Depends(get_mongo_db),
+    user: User = Depends(get_current_active_user),
+):
+    """
+    Create a JGI Sample.
+    """
+
+    # check for permissions first
+    check_can_run_wf_file_staging_endpoints(user)
+    # check if record with same task_id already exists
+    existing = mdb["wf_file_staging.jgi_samples"].find_one(
+        {"jdp_file_id": jgi_in.jdp_file_id}
+    )
+    if existing is not None:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"JGI sample with jdp_file_id {jgi_in.jdp_file_id} already exists.",
+        )
+
+    sample_dict = jgi_in.model_dump(exclude_unset=True)
+    mdb["wf_file_staging.jgi_samples"].insert_one(sample_dict)
+    return sample_dict
+
+
+@router.get(
+    "/wf_file_staging/jgi_samples",
+    response_model=ListResponse[JGISample],
+    response_model_exclude_unset=True,
+)
+def list_jgi_samples(
+    req: Annotated[ListRequest, Query()],
+    mdb: Database = Depends(get_mongo_db),
+):
+    r"""
+    Retrieves JGI Sample records that match the specified filter criteria. Uses Mongo-like filters.
+    """
+
+    rv = list_resources(req, mdb, "wf_file_staging.jgi_samples")
+    rv["resources"] = [strip_oid(d) for d in rv["resources"]]
+    return rv
+
+
+@router.patch("/wf_file_staging/jgi_samples/{jdp_file_id}", response_model=JGISample)
+def update_jgi_samples(
+    jdp_file_id: str,
+    sample_patch: JGISample,
+    mdb: Database = Depends(get_mongo_db),
+    user: User = Depends(get_current_active_user),
+):
+    """
+    Update a JGI Sample record by the jdp_file_id.
+    """
+    # check for permissions first
+    check_can_run_wf_file_staging_endpoints(user)
+
+    if jdp_file_id != sample_patch.jdp_file_id:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="jdp_file_id in path and body must match.",
+        )
+
+    doc = raise404_if_none(
```

Comment:
I'd prefer this name be more specific since this endpoint deals with multiple documents (er... states of a document). To mirror the other variable's name, it could be:

```suggestion
    doc_sample_original = raise404_if_none(
```